bot start pack.

artich02 opened the pull request:
# Pull Request

## Что сделано
Пример:
- Добавлен базовый бот @one_hundred_eight_bot
- Функции: одиночная дуэль, дуэль по коду, дуэль по поиску, создание вопросов gigachat

## Как проверить
Создание образа:
```
docker build -t quiz-bot .  
```
Запуск образа:
```
docker run -d --name quiz-bot --restart unless-stopped quiz-bot
```
Остановка образа:
```
docker stop quiz-bot
```
альтернативно запуск main.py и вход в @one_hundred_eight_bot в ТГ

artich02 committed:
бот 1 трай

artich02 committed:
изменение версии питона в докере

artich02 committed:
убрал мусор из зависимостей

WholeHorse commented on bot/bot.py:
```diff
@@ -0,0 +1,398 @@
+import json
+from gigachat import GigaChat  # pip install gigachat
+
+# ==================== ТОКЕНЫ ====================
```
Ну тут и так понятно, нужно использовать python-dotenv и os.getenv, а токены в .env файле хранить

WholeHorse commented on bot/bot.py:
````diff
@@ -0,0 +1,398 @@
+        if raw.startswith("```json"):
+            raw = raw[7:-3]
+        elif raw.startswith("```"):
+            raw = raw[3:-3]
````
Не совсем надежно, нейросетки могут вернуть например "```json " (с пробелом) и будет проблема с парсингом.
Можно попробовать использовать регулярные выражения или библиотеку для поиска json в тексте

WholeHorse commented on bot/bot.py:
```diff
@@ -0,0 +1,398 @@
+    }
+
+# ==================== ХРАНИЛИЩА ====================
+user_data = {}
+rooms = {}
+quick_queue = []
```
Сейчас это глобальные словари, при перезапуске кода (например падение сервера) все текущие игры оборвутся, а комнаты исчезнут. Для MVP ок, но можно добавить коммент с TODO, чтоб использовать Redis или БД для хранения состояния активных игр

WholeHorse commented on bot/bot.py:
```diff
@@ -0,0 +1,398 @@
+        pass
+
+# ==================== БАЗА ЛИДЕРОВ ====================
+conn = sqlite3.connect('leaders.db', check_same_thread=False)
```
Telebot работает в многопоточном режиме. SQLite допускает чтение из разных потоков, но одновременная запись может вызвать ошибку Database is locked или повредить файл.
В идеале, потом нужно будет перейти на Postgres

WholeHorse commented on bot/bot.py:
```diff
@@ -0,0 +1,398 @@
+            print("Новый квиз успешно получен от GigaChat")
+            return data
+    except Exception as e:
+        print(f"GigaChat error: {e}")
```
Для логирование лучше использовать не print, а библиотеку logging (во всем коде)
Например:
```python
import logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

logger.error(f"GigaChat error: {e}")
```

WholeHorse commented on bot/bot.py:
```diff
@@ -0,0 +1,398 @@
+    try:
+        bot.delete_message(chat_id, message_id)
+    except:
+        pass
```
Лучше указать конкретные исключения или сделать логирование перед pass

WholeHorse reviewed (commented):
В целом хорошо, есть некоторые критические проблемы, некоторые средние, а некоторые пока ок, а к MVP исправить бы

artich02 committed:
добавлены мероприятия
исправления по ревью
добавлены информационные плашки
исправлены ошибки в выборе правльного ответа
